docs/source/tensor_attributes.rst

```diff
@@ -34,6 +34,37 @@ Boolean                    ``torch.bool``                                ``torch
 To find out if a :class:`torch.dtype` is a floating point data type, the property :attr:`is_floating_point`
 can be used, which returns ``True`` if the data type is a floating point data type.
 
+For arithmetic operations we allow casting a tensor to an output tensor of another dtype, except:
```

Comment:
this only documents the cases where the output is explicitly specified -- what about cases with no outputs?